qiskit_algorithms/optimizers/aqgd.py

```diff
@@ -57,6 +57,7 @@ def __init__(
         momentum: float | list[float] = 0.25,
         param_tol: float = 1e-6,
         averaging: int = 10,
+        max_evals_grouped: int = 1,
```

Comment:
This should be added to the Args: docstring below. And I think we should note that max_evals_grouped will max out at 3 such that even if a larger number is passed the max evals that can be grouped is 3.

Reply:
I have just two short question regarding the second part. Maybe I misunderstood the question, but shouldn't it max out at 2*n+1, if n is the dimension of x0? And shouldn't it then also be specified for the method `gradient_num_diff` in the class [Optimizer(ABC)](https://docs.quantum.ibm.com/api/qiskit/qiskit.algorithms.optimizers.Optimizer)?

Reply:
Regarding the first part. Makes sense. I will do so 👍

Reply:
>  but shouldn't it max out at 2*n+1

Yes, you are right. I was confusing things. SPSA does have limit since it perturbs all parameters together for + and - but for no limit in this case its fine as is - whatever the limit its down to the optimizer subclass - normally it scales some way based on num parameters (dimension of point) like this.

Reply:
I added the changes. Please take a look if that fits.